### ISSUE TITLE
Performance of deeplab v3+ model in cityscape model?

### ISSUE THREAD

**Issue by YjyJeff** [tensorflow/models#4214]
### System information
- **What is the top-level directory of the model you are using**: deeplab
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**:1.8.0
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version**: 9/7
- **GPU model and memory**: P40/ 22G
- **Exact command to reproduce**:

Hey, I want to reproduce the result of deeplab v3+ in CityScape dataset. Instead of using the code in deeplab directory, I am using the  [keras version](https://github.com/bonlime/keras-deeplab-v3-plus) .  I download the pretrained weights from the deeplab directory and use keras model to load it. In my view,  this should have the same result as the tensorflow version(I use tensorflow as backend). 

Actually, the performance looks very good :). However, when I am using [cityscape evaluation script](https://github.com/mcordts/cityscapesScripts). I only got 74.77% class mIOU and 86.31% category mIOU in evaluation dataset. It still has difference with the benchmark provided in [CityScape dataset website](https://www.cityscapes-dataset.com/benchmarks/)(82.1% class mIOU and 92% category mIOU). Does the test dataset in their evaluation server will have more accuracy?

I know the trainID and evaluation ID is different, therefore I translate it back. Here is the code how I use keras to predict:

     reverse_map = {0:7, 1:8, 2:11,3:12,  4:13,  5:17,6:19, 7:20, 8:21,9:22,10:23,11:24,12:25,13:26, 14:27,15:28,16:31,17:32,18:33}
     seg_predict = model.predict(batch_feature_val)
     seg_predict = np.argmax(seg_predict, axis = 3)
     for filename, predict in zip(batch_img_filenames, seg_predict):
            predict_filename = result_dir + filename.decode('utf-8') + "_predict.png"
            predict = np.vectorize(reverse_map.get)(predict)
            im = Image.fromarray(predict.astype('uint8'))
            im.save(predict_filename)

for each image in validation dataset, I will generate a predicted png file so that it can be evaluated by the provided script. 

Anything wrong with my code? Could anyone reproduce the 82.1% class mIOU accuracy in validation dataset? Thanks in advance!

**Comment by aquariusjay** [comment]
To get the 82.1% performance (on **test** set), you need to further train the model on all the fine + coarse annotations.

**Comment by YjyJeff** [comment]
Haha, got it. What's more, the accuracy in the validation dataset still has difference with (given benchmark)[https://github.com/tensorflow/models/blob/master/research/deeplab/g3doc/model_zoo.md] (78.79% class mIOU) . Would the keras version goes wrong?

**Comment by aquariusjay** [comment]
You need to use large batch size to train the model, as batch norm is important to attain the performance.

**Comment by YjyJeff** [comment]
Thanks a lot! I will have a try.

**Comment by lxtGH** [comment]
Hi @YjyJeff  what is your result ?

**Comment by kshitijagrwl** [comment]
Hi @YjyJeff how many steps did it take to achieve that resultm and what was the learning rate? 
I'm hovering at ~56%mIoU after roughly 240,000 steps with batch size of 8.

**Comment by baoruxiao** [comment]
Hi @YjyJeff, I got similar results as yours by using official tensorflow implementation, I got 74.7% mIoU.
